.gitlab-ci.yml

```diff
@@ -0,0 +1,203 @@
+stages:
+  - build
+  - deploy
+
+.artifacts: &artifacts
+  artifacts:
+    paths:
+      - build/
+
+.cache: &cache
+  cache:
+    key: "$CI_BUILD_NAME"
+    paths:
+      - cache/
+
+
+#================================ DEBIAN-BASED ================================
+
+.build_rc_package_deb: &build_rc_package_deb
+  stage: build
+  script:
+    - mkdir build || true # allowed to fail
+    - cd build
+    - cmake .. -DWITH_SERVER=1 -DCMAKE_BUILD_TYPE=Release -DCPACK_GENERATOR=DEB
+    - make package -j2
+
+.build_debug_package_deb: &build_debug_package_deb
+  stage: build
+  script:
+    - mkdir build || true # allowed to fail
+    - cd build
+    - cmake .. -DWITH_SERVER=1 -DCMAKE_BUILD_TYPE=Debug -DCPACK_GENERATOR=DEB
+    - make package -j2
+
+.deb-artifacts: &artifacts_deb
+  artifacts:
+    paths:
+      - build/*.deb
+      - build/CMakeFiles/*.log
+    when: always
+
+#----------------------------------- UBUNTU -----------------------------------
+
+.requirements_16xx: &install_requirements_16xx
+  before_script:
+    - apt-get -o dir::cache::archives="cache" update -qq
+    - apt-get -o dir::cache::archives="cache" install -y build-essential g++ cmake git
+    - apt-get -o dir::cache::archives="cache" install -y libprotobuf-dev protobuf-compiler
+    - apt-get -o dir::cache::archives="cache" install -y qt5-default qttools5-dev qttools5-dev-tools
+    - apt-get -o dir::cache::archives="cache" install -y qtmultimedia5-dev libqt5multimedia5-plugins
+    - apt-get -o dir::cache::archives="cache" install -y libqt5svg5-dev libqt5sql5-mysql
+    - apt-get -o dir::cache::archives="cache" install -y libqt5websockets5-dev
```

Comment:
Mostly so it looks nice in the yml. Splitting it up doesn't have any real impact on performance.